packages/admin/admin/src/common/toolbar/DataGridToolbar.tsx

```diff
@@ -0,0 +1,41 @@
+import { ComponentsOverrides } from "@mui/material";
+import { Theme, useThemeProps } from "@mui/material/styles";
+import * as React from "react";
+
+import { createComponentSlot } from "../../helpers/createComponentSlot";
+import { ThemedComponentBaseProps } from "../../helpers/ThemedComponentBaseProps";
+import { Toolbar, ToolbarProps } from "./Toolbar";
+
+export type DataGridToolbarClassKey = "root";
+
+export type DataGridToolbarProps = Omit<ToolbarProps, "slotProps" | "scopeIndicator" | "hideTopBar" | "hideBottomBar"> &
+    ThemedComponentBaseProps<{
+        root: typeof Toolbar;
+    }>;
+
+const Root = createComponentSlot(Toolbar)<DataGridToolbarClassKey>({
+    componentName: "DataGridToolbar",
+    slotName: "root",
+})();
+
+export const DataGridToolbar = (inProps: DataGridToolbarProps) => {
+    const props = useThemeProps({ props: inProps, name: "CometAdminDataGridToolbar" });
+    return <Root {...props} hideTopBar />;
+};
```

Comment:
@jamesricky I'm actually not sure if this is the way to go for a pure wrapper component like this. Do I still need a root slot? What do you think?

Reply:
nvm with the `density` it's definitely needed. But I'm still not sure if I've done it correctly

Reply:
Generally this is correct, the root slot is still needed. 
However you could probably just remove the `root` inside the `ThemedComponentBaseProps` generic, as mentioned here: https://github.com/vivid-planet/comet/pull/2050#discussion_r1598033860